src/mir/analysis_context.rs

```diff
@@ -90,33 +91,37 @@ impl<'tcx, 'compilation> AnalysisContext<'tcx, 'compilation> {
                 if def_kind == DefKind::Fn || def_kind == DefKind::AssocFn {
                     let item_name = tcx.item_name(local_def_id.to_def_id());
                     if item_name.to_string() == *entr_func {
-                        entry_fn_def_id = Some(local_def_id.to_def_id());
+                        entry_fn_def_ids.insert(local_def_id.to_def_id());
                     }
                 }
             }
+        } else {
+            // If `entry_func` is not specified and the local crate is a library,
+            // then we add all effectively public (i.e. can be called by library users)
+            // functions to the `entry_funcs`.
+            entry_fn_def_ids.extend(&lib_entry_funcs(tcx));
```

Comment:
The entry point for the analysis can be specified using either the `entry_func` or `entry_def_id` option. I think the `entry_fn_def_ids` should include all the library entry functions when neither option is set.

We need to check if the local crate is a library.